src/libraries/Common/src/System/Security/Cryptography/MLKemImplementation.Windows.cs

```diff
@@ -0,0 +1,257 @@
+                    blobHeaderSize +
+                    parameterSetMarshalLength +
+                    key.Length;
+
+                byte[]? rented = null;
+                Span<byte> buffer = (uint)blobSize <= MaxKeyStackSize ?
+                    stackalloc byte[MaxKeyStackSize] :
+                    (rented = CryptoPool.Rent(blobSize));
+
+                try
+                {
+                    buffer.Clear();
+
+                    unsafe
+                    {
+                        fixed (byte* pBuffer = buffer)
+                        {
+                            BCRYPT_MLKEM_KEY_BLOB* blob = (BCRYPT_MLKEM_KEY_BLOB*)pBuffer;
+                            blob->dwMagic = kind;
+                            blob->cbParameterSet = (uint)parameterSetMarshalLength;
+                            blob->cbKey = (uint)key.Length;
+                        }
+                    }
+
+                    // This won't write the null byte, but we zeroed the whole buffer earlier.
+                    Encoding.Unicode.GetBytes(parameterSet, buffer.Slice(blobHeaderSize));
+                    key.CopyTo(buffer.Slice(blobHeaderSize + parameterSetMarshalLength));
+                    string blobKind = PqcBlobHelpers.MLKemBlobMagicToBlobType(kind);
+                    return Interop.BCrypt.BCryptImportKeyPair(
+                        s_algHandle,
+                        blobKind,
+                        buffer.Slice(0, blobSize));
+                }
+                finally
+                {
+                    if (rented is not null)
+                    {
+                        CryptoPool.Return(rented, blobSize);
+                    }
+                }
+            }
+        }
+
+        private void ExportKey(KeyBlobMagicNumber kind, Span<byte> destination)
+        {
+            string blobKind = PqcBlobHelpers.MLKemBlobMagicToBlobType(kind);
+            ArraySegment<byte> exported = Interop.BCrypt.BCryptExportKey(_key, blobKind);
+
+            try
+            {
+                Span<byte> exportedSpan = exported;
+
+                unsafe
+                {
+                    fixed (byte* pExportedSpan = exportedSpan)
+                    {
+                        BCRYPT_MLKEM_KEY_BLOB* blob = (BCRYPT_MLKEM_KEY_BLOB*)pExportedSpan;
+
+                        if (blob->dwMagic != kind)
+                        {
+                            Debug.Fail("dwMagic is not expected value");
+                            throw new CryptographicException();
+                        }
+
+                        int blobHeaderSize = Marshal.SizeOf<BCRYPT_MLKEM_KEY_BLOB>();
+                        int keySize = checked((int)blob->cbKey);
+                        int paramSetSize = checked((int)blob->cbParameterSet);
+                        string paramSet = Marshal.PtrToStringUni((nint)(pExportedSpan + blobHeaderSize))!;
+                        string expectedParamSet = PqcBlobHelpers.GetMLKemParameterSet(Algorithm);
```

Comment:
If we wanted, we could avoid the string by doing something like

```C#
ReadOnlySpan<byte> paramSetBytes = new(pExportedSpan + blobHeaderSize, paramSetSize);
ReadOnlySpan<char> paramSetChars = MemoryMarshal.Cast<byte, char>(paramSetBytes);
ReadOnlySpan<char> paramSetCharsNoTerm = paramSetChars[0..^2];

if (!paramSetCharsNoTerm.SequenceEquals(expectedParamSet) || paramSetChars[^1] != '\0' || keySize != destination.Length))
{
   throw!
}
```

But I'm not sure it's important.

I guess we could write the strcmp helper once and get the benefit without making the code locally ugly.

Reply:
The right way to do this is `MemoryMarshal.CreateReadOnlySpanFromNullTerminated`, but that doesn't have .NET Standard availability.

I generally try to stay clear of "`reinterpret_cast`: .NET edition" but it doesn't look too bad to do it.